src/core/nixl_agent.cpp

```diff
@@ -38,6 +39,21 @@
         }                                                 \
     } while (0)
 
+/* ERROR log prefixed with current function name and a colon */
+#define NIXL_ERROR_FUNC NIXL_ERROR << __FUNCTION__ << ": "
+
+#define SERIALIZE_ERR(ret)                         \
+    do {                                           \
+        NIXL_ERROR_FUNC << "serialization failed"; \
+        return ret;                                \
+    } while (0)
+
+#define DESERIALIZE_ERR()                            \
+    do {                                             \
+        NIXL_ERROR_FUNC << "deserialization failed"; \
+        return NIXL_ERR_MISMATCH;                    \
+    } while (0)
```

Comment:
The serialize errors should be moved to serializer class, and NIXL_ERROR_FUNC should be moved to the common log.h header